File: backend/src/main/java/lan/dk/podcastserver/manager/worker/noop/NoOpFinder.java
package lan.dk.podcastserver.manager.worker.noop;

import lan.dk.podcastserver.entity.Podcast;
import lan.dk.podcastserver.manager.worker.Finder;
import javax.validation.constraints.NotEmpty;

/**
 * Created by kevin on 08/03/2016 for Podcast Server
 */
public class NoOpFinder implements Finder {
    @Override
    public Podcast find(String url) {
        return Podcast.DEFAULT_PODCAST;
    }

    @Override
    public Integer compatibility(@NotEmpty String url) {
        return -1;
    }
}

File: backend/src/main/kotlin/com/github/davinkevin/podcastserver/manager/worker/noop/NoOpFinder.kt
package com.github.davinkevin.podcastserver.manager.worker.noop

import lan.dk.podcastserver.entity.Podcast
import com.github.davinkevin.podcastserver.manager.worker.Finder

/**
 * Created by kevin on 08/03/2016 for Podcast Server
 */
class NoOpFinder : Finder {
    override fun find(url: String): Podcast = Podcast.DEFAULT_PODCAST
    override fun compatibility(url: String?) = -1
}
